dali/kernels/imgproc/resample/resampling_internal_test.cu

```diff
@@ -0,0 +1,470 @@
+      filter, filter.support());
+    cudaDeviceSynchronize();
+  }
+
+  cv::Mat out;
+  out.create(H, outW, CV_8UC3);
+  auto img_out_cpu = view_as_tensor<uint8_t, 3>(out);
+  auto img_ref_cpu = view_as_tensor<uint8_t, 3>(cv_ref);
+  copy(img_out_cpu, img_out);  // NOLINT
+  cudaDeviceSynchronize();
+  EXPECT_NO_FATAL_FAILURE(Check(img_out_cpu, img_ref_cpu, EqualEps(1))) <<
+  [&]() {
+    cv::Mat diff;
+    cv::absdiff(out, cv_ref, diff);
+    cv::imwrite("resample_horz_dif.png", diff);
+    return "Test failed. Absolute difference image saved to resample_horz_dif.png";
+  }();
+}
+
+TEST(Resample, VerticalGaussian) {
+  auto cv_img = testing::data::image("imgproc_test/checkerboard.png");
+  auto cv_ref = testing::data::image("imgproc_test/ref_out/resample_vert.png");
+  ASSERT_FALSE(cv_img.empty()) << "Couldn't load the image";
+  TensorView<StorageCPU, const uint8_t, 3> img;
+  ASSERT_NO_FATAL_FAILURE((img = view_as_tensor<uint8_t>(cv_img)));
+
+  int channels = img.shape[2];
+  ASSERT_EQ(channels, 3);
+  int H = img.shape[0];
+  int W = img.shape[1];
+  int outH = H / 2;
+  auto gpu_mem_in = memory::alloc_unique<uint8_t>(AllocType::GPU, img.num_elements());
+  auto gpu_mem_out = memory::alloc_unique<uint8_t>(AllocType::GPU, outH * W * channels);
+
+  TensorView<StorageGPU, uint8_t, 3> img_in, img_out;
+  img_in = { gpu_mem_in.get(), img.shape };
+  img_out = { gpu_mem_out.get(), { outH, W, channels } };
+
+  copy(img_in, img);  // NOLINT
+
+  auto filters = GetResamplingFilters(0);
+  ResamplingFilter filter = (*filters)[1];
+
+  int radius = 40;
+  filter.rescale(2*radius+1);
+
+  for (int i = 0; i < 100; i++) {
+    ResampleVertTestKernel<<<1, dim3(32, 24), ResampleSharedMemSize>>>(
+      img_out.data, W*channels, outH, img_in.data, W*channels, W, H, channels,
+      filter, filter.support());
+    cudaDeviceSynchronize();
+  }
+
+  cv::Mat out;
+  out.create(outH, W, CV_8UC3);
+  auto img_out_cpu = view_as_tensor<uint8_t, 3>(out);
+  auto img_ref_cpu = view_as_tensor<uint8_t, 3>(cv_ref);
+  copy(img_out_cpu, img_out);  // NOLINT
+  cudaDeviceSynchronize();
+  EXPECT_NO_FATAL_FAILURE(Check(img_out_cpu, img_ref_cpu, EqualEps(1))) <<
+  [&]() {
+    cv::Mat diff;
+    cv::absdiff(out, cv_ref, diff);
+    cv::imwrite("resample_vert_dif.png", diff);
+    return "Test failed. Absolute difference image saved to resample_hv_dif.png";
+  }();
+}
+
+class ResamplingTest : public ::testing::Test {
+ public:
+  enum BlockConfig {
+    BlockPerImage = 0,
+    BlockPerSpan = 1,
+  };
+
+  void SetOutputSize(int w, int h) {
+    out_w_ = w;
+    out_h_ = h;
+    scale_x_ = static_cast<double>(out_w_) / InputWidth();
+    scale_y_ = static_cast<double>(out_h_) / InputWidth();
+    fixed_size_ = true;
+  }
+  void SetScale(double sx, double sy) {
+    scale_x_ = sx;
+    scale_y_ = sy;
+    out_w_ = sx * InputWidth();
+    out_h_ = sy * InputHeight();
+    fixed_size_ = false;  }
```

Comment:
break line here

Reply:
I wonder how it passed the linter O_o.

Reply:
the linter doesn't seem very smart when it comes to indentations and stuff like that